src/main.rs

```diff
@@ -104,6 +105,19 @@ async fn main() -> Result<(), anyhow::Error> {
     let manifest = load_manifest_path(manifest_path.join("Cargo.toml"))?;
     let package = manifest.package.unwrap();
 
+    let plain_version_rx = regex::Regex::new(r"^\d+[.]\d+[.]\d+(-\w+([.]\d+)?)?$").unwrap();
```

Comment:
Well, cargos version syntax lets you do crazy things and I think handling that regex in this crate would be hell to maintain different use cases. Wouldn’t [this crate](https://github.com/dtolnay/semver) be the perfect match for this?